orchestration/dagster_orchestration/hca_orchestration/support/typing.py

```diff
@@ -12,3 +14,13 @@ class HcaScratchDatasetName(str):
 
 
 make_python_type_usable_as_dagster_type(HcaScratchDatasetName, DagsterString)
+
+
+class MetadataType(str):
```

Comment:
This more generic type and fanout result type works for file and nonfile metadata, replaces the existing ones (that are now deleted)